docs/reference/builder.md

```diff
@@ -1181,12 +1183,18 @@ For example, consider this Dockerfile:
 3 RUN echo $CONT_IMG_VER
 ```
 
-If you specify `--build-arg CONT_IMG_VER=<value>` on the command line the
-specification on line 2 does not cause a cache miss; line 3 does cause a cache
-miss. The definition on line 2 has no impact on the resulting image. The `RUN`
-on line 3 executes a command and in doing so defines a set of environment
-variables, including `CONT_IMG_VER`. At that point, the `ARG` variable may
-impact the resulting image, so a cache miss occurs.
+```
+1 FROM ubuntu
+2 ARG CONT_IMG_VER
+3 RUN echo hello
+```
+If you specify `--build-arg CONT_IMG_VER=<value>` on the command line, in both
+cases, the specification on line 2 does not cause a cache miss; line 3 does cause
+a cache miss. The definition on line 2 has no impact on the resulting image. The
+`RUN` on line 3 executes a command and in doing so defines a set of environment
+variables, including `CONT_IMG_VER` if it has a value associated with it. At
+that point, the `ARG` variable may impact the resulting image, so a cache miss
+occurs.
```

Comment:
I think you need to add a few things here. In the 2nd example you never use CONT_IMG_VER in the RUN command but you'll get a cache miss the same as the first example. You need to make it clear that the presence of $CONT_IMG_VER on the RUN command will have no impact on whether there's a cache hit or not.

Also, you make it sound like we'll always have a cache miss, and this isn't true.  We'll only have one IF CONT_IMG_VER is different from a previous build.